### PR TITLE
CUMULUS-802 graceful task termination

### PR DESCRIPTION
**Pull request by yjpa7145**
* Update the ECS cluster to use the [Amazon ECS-optimized AMI](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-optimized_AMI.html).  This should fix [CUMULUS-966](https://bugs.earthdata.nasa.gov/browse/CUMULUS-966) as well.
* Use the [new version](https://github.com/nasa/cumulus-ecs-task/releases/tag/v1.2.5) of the cumulus-ecs-task docker image, which gracefully terminates tasks.

### CODE REVIEW

File: packages/deployment/app/config.yml [hunk 1]
```diff
@@ -98,7 +98,8 @@ default:
 
   ecs:
     restartTasksOnDeploy: false
-    amiid: ami-6944c513
+    container_stop_timeout: 2m
+    amiid: ami-007571470797b8ffa
```
Comment (abarciauskas-bgse) [inline packages/deployment/app/config.yml]: Don't the DAAC's have to use the NGAP base ami? Can we ask it's built from this base ami?

File: example/spec/parallel/asyncOperations/asyncOperationRunnerSpec.js [hunk 2]
```diff
@@ -84,7 +84,7 @@ describe('The AsyncOperation task runner', () => {
       expect(parsedOutput.message).toContain('Function not found');
     });
     // Disabled pending resolution of CUMULUS-966
```
Comment (markdboyd) [inline example/spec/parallel/asyncOperations/asyncOperationRunnerSpec.js]: Seems like we could remove these comments too?

Comment (yjpa7145) [reply example/spec/parallel/asyncOperations/asyncOperationRunnerSpec.js]: Fixed in 6787740b6

**Comment by yjpa7145** [comment]
**Test plan**

- [ ] EC2 instance starts up and joins cluster
- [ ] When scaling out, the new instance joins the cluster
- [ ] The correct ECS_ENGINE_TASK_CLEANUP_WAIT_DURATION is set in ecs.config
- [ ] The correct DOCKER_STORAGE_OPTIONS are set in /etc/sysconfig/docker-storage
- [ ] If ecs.docker.storageDriver = "devicemapper", correct dm.basesize set in /etc/sysconfig/docker
- [ ] If ecs.docker.registry = "dockerhub", correct username / password configured in ecs.config and able to fetch images from dockerhub
- [ ] If ecs.efs.mount
  - [ ] nfs-utils package is installed
  - [ ] mount dir is created
  - [ ] EFS filesystem is mounted properly
- [ ] When the service size is scaled down, tasks shut down gracefully

File: packages/deployment/app/cloudformation.template.yml [hunk 3]
```diff
@@ -981,14 +890,48 @@ Resources:
...
-        "Fn::Base64":
-          Fn::Join:
-            - ""
-            - - "#cloud-boothook\n"
-              - "\n"
-              - "yum install -y aws-cfn-bootstrap\n"
-              - Fn::Sub: "/opt/aws/bin/cfn-init -v --stack ${AWS::StackName} --resource CumulusContainerInstanceLaunch --region ${AWS::Region} -c MountConfig\n"
-              - Fn::Sub: "/opt/aws/bin/cfn-signal -e $? --stack ${AWS::StackName} --resource CumulusECSAutoScalingGroup --region ${AWS::Region}"
+        Fn::Base64:
+          Fn::Sub: |
```
Comment (Jkovarik) [inline packages/deployment/app/cloudformation.template.yml]: I don't see any reason this shouldn't work in CW, presuming the test plan from #901 and the kes CF merge integration test pass.

**Comment by yjpa7145** [comment]
Completed the [test plan](https://github.com/nasa/cumulus/pull/901#issuecomment-479059438), so fingers crossed.

**Pull request merged**